pkg/git/repository.go

```diff
@@ -168,18 +190,26 @@ func (di *defaultRepositoryImpl) cherryPickMergeCommit(
 // function should work with commits, tags and other objects, but currently it only
 // works with
 func (di *defaultRepositoryImpl) checkout(client *gogit.Repository, opts *RepoOptions, refName string) error {
-	tree, err := client.Worktree()
-	if err != nil {
-		return errors.Wrap(err, "getting repository worktree")
-	}
-	// TODO: Check if reference is a hash and use that
-	if err := tree.Checkout(
-		&gogit.CheckoutOptions{
-			Branch: plumbing.NewBranchReferenceName(refName),
-		},
-	); err != nil {
-		return errors.Wrapf(err, "checking out %s", refName)
+	logrus.Infof("Checking out branch %s", refName)
+	// Switch to the sourceBranch, this ensures it exists and from there we branch
+	if err := command.NewWithWorkDir(
+		opts.Path, gitCommand, "checkout", refName).RunSilentSuccess(); err != nil {
+		return errors.Wrapf(err, "switching to source branch %s", refName)
 	}
+	/*
```

Comment:
do we need to keep this? if so maybe add a more descriptive comment

Reply:
I wanted to write this one in pure go but it turned out very problematic. I added the bash version a single commit we can revert in the future to continue the previous version